oolab/src/main/java/agh/ics/oop/Animal.java

```diff
@@ -19,6 +22,7 @@ public Animal(IWorldMap map, Vector2d position) {
         this.direction = MapDirection.NORTH;
         this.position = position;
         this.map = map;
+        addObserver((IPositionChangeObserver) map);
```

Comment:
Jeśli każda mapa rozszerzałaby ten interfejs to nie musielibyśmy rzutować -> interfejsy też mogą rozszerzać inne interfejsy.